### PR TITLE
Replace google with leaflet/mapbox

### PR DESCRIPTION
**Pull request by toddjordan**

### CODE REVIEW

File: app/components/location-map.js [hunk 1]
```diff
@@ -2,12 +2,14 @@ import { inject as service } from '@ember/service';
...
+  classNames: ['map-container'],
+  mapElement: service(),
 
   didInsertElement() {
     this._super(...arguments);
-    let location = this.location;
-    let mapElement = this.maps.getMapElement(location);
-    this.$('.map-container').append(mapElement);
+    this.mapElement.getMapElement(this.location).then((mapElement) => {
+      this.$().append(mapElement);
```
Comment (sivakumar-kailasam) [inline app/components/location-map.js]:
```suggestion
      this.element.appendChild(mapElement);
```

Comment (sivakumar-kailasam) [reply app/components/location-map.js]: Since we weren't using jQuery APIs elsewhere in the PR

Comment (toddjordan) [reply app/components/location-map.js]: fixed

**Comment by jenweber** [comment]
Possible later PRs:
- I think we should add the key instructions to the README for the repo. I can help with this :)

**Review by jenweber — CHANGES_REQUESTED** [review]
For the individual listing pages, there appears to be a clash between the flex boxes and the "normal" % width. This shows on smallish screens (13 inch macbook) and as you resize.

screen shot 2018-11-10 at 3 32 14 pm

I'm still looking through the copy paste together with the guides PR, but everything else seems good!

**Comment by toddjordan** [comment]
@jenweber I see what you mean here: 
super-bug

I think we can fix using a min width

**Comment by toddjordan** [comment]
Hi @jenweber I added some fixes to the PR.  Also noticed a couple of things.  This new PR requires version 3 of ember-cli-tutorial-style which I haven't yet published.  I didn't want to publish it and mess up people using the current tutorial.  If you want to test with that stylesheet you can the ember-tutorial.css from https://github.com/ember-learn/super-rentals/blob/973fe98bf01828197a26d081adee40edfb800de4/vendor/ember-tutorial.css into your own vendor directory.  Also I found a place where I didn't update the markup which would cause some layout issue with the listings.  I've since updated.  Now you should see some better behavior on your 13 inch screen.  When you are ok with this merging I will publish the tutorial css right before merge.

**Pull request merged**